src/mixins/legal-api-mixin.ts

```diff
@@ -158,53 +160,135 @@ export default class LegalApiMixin extends Mixins(FilingTemplateMixin) {
   }
 
   /**
-   * Fetch business data for current business.
-   * @param datatype The type of data to request.
-   * @returns a promise to return the business base info or of the specified data type.
+   * Fetches the business info of the current business.
+   * @returns a promise to return the data
    */
-  async getBusinessData (datatype: string = null): Promise<any> {
-    if (!this.getBusinessId) throw new Error('Invalid parameter \'businessIdentifier\'')
+  async fetchBusinessInfo (): Promise<BusinessInformationIF> {
+    if (!this.getBusinessId) throw new Error('Invalid business id')
 
-    let url = `businesses/${this.getBusinessId}`
-    if (datatype) {
-      url += `/${datatype}`
-    }
+    const url = `businesses/${this.getBusinessId}`
 
     return axios.get(url)
       .then(response => {
-        if (response && response.data) {
+        if (response?.data) {
+          return response.data.business
+        } else {
+          // eslint-disable-next-line no-console
+          console.log('fetchBusinessInfo() error - invalid response =', response)
+          throw new Error('Invalid API response')
+        }
+      })
+  }
+
+  /**
+   * Fetches the name translations of the current business.
+   * @returns a promise to return the data
+   */
+  async fetchNameTranslations (): Promise<NameTranslationIF[]> {
```

Comment:
Didn't like the one method with parameters to make the requests?

Reply:
Yes, I liked it... but it didn't work in all cases, since different network calls need to return different things.

Reply:
For sure, managing types of response objects was lacking in the previous implementation.